internal/pkg/service/appsproxy/dataapps/api/rule.go

```diff
@@ -32,15 +32,20 @@ type Rule struct {
 	AuthRequired *bool         `json:"authRequired"`
 }
 
-func (r *Rule) RegisterHandler(mux *http.ServeMux, handler http.Handler) error {
+func (r *Rule) Match(req *http.Request) (bool, error) {
```

Comment:
Do you need `bool`. Just by inspecting i see, `err == nil` is exactly what bool matches.

Reply:
No, there is a difference, I am adding tests for it. 

Error means, the rule is invalid.

`false` means that the rude doesn't match the path.